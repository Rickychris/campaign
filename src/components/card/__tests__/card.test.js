import React from "react";
import ReactDom from "react-dom";
import { render, cleanup } from "@testing-library/react";
import Card from "../card";

describe("Card tests", () => {
  beforeEach(cleanup);
  const cardProps = {
    id: "unique ID",
    campaignName: "Email Campaign Test",
    creatorName: "Alvis Monk",
    createdAt: 1626289185097,
    lastModifiedAt: 1626289185097,
    status: "active",
  };
  it("renders the Card", () => {
    const div = document.createElement("div");
    ReactDom.render(<Card cardData={cardProps}></Card>, div);
  });

  it("renders the Card with status correctly", () => {
    const { getByTestId } = render(<Card cardData={cardProps} />);
    expect(getByTestId("card")).toHaveTextContent("Status : active");
  });

  it("matches snapshot", () => {
    const { getByTestId } = render(<Card cardData={cardProps} />);
    expect(getByTestId("card")).toMatchSnapshot();
  });
});
