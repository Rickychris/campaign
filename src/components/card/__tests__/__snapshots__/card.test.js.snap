// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests matches snapshot 1`] = `
<div
  class="root"
  data-testid="card"
>
  <p>
    Created By : Alvis Monk
  </p>
  <p>
    Thu Jul 15 2021
  </p>
  <p>
    Status : active
  </p>
  <div>
    <p>
      delete
    </p>
    <p>
      Send
    </p>
  </div>
</div>
`;
